api/rest/swagger.yaml

```diff
@@ -0,0 +1,796 @@
+
+components:
+  schemas: 
+    Cid:
+      type: object
+      properties:
+        /:
+          type: string
+
+    ConnectGraph:
+      type: object
+      properties:
+        ClusterID:
+          type: string
+        ipfs_links:
+          type: object
+          additionalProperties:
+              type: array
+              items:
+                type: string
+        cluster_links:
+          type: object
+          additionalProperties:
+              type: array
+              items:
+                type: string      
+        cluster_to_ipfs:
+          type: object
+          additionalProperties:
+              type: string
+          
+    Peer:
+      type: object
+      properties:
+        peername:
+          type: string
+        addresses:
+          type: array
+          items:
+            type: string
+        ipfs:
+          type: object
+          properties:
+            addresses: {}
+            error:
+              type: string
+        cluster_peers_addresses:
+          type: array
+          items:
+            type: string
+        commit:
+          type: string
+        rpc_protocol_version:
+          type: string
+        cluster_peers:
+          type: array
+          items:
+            type: string
+        id:
+          type: string
+        error:
+          type: string
+        version:
+          type: string
+    
+    PinOptions:
+      type: object
+      properties:
+        reference: {}
+        metadata: {}
+        allocations:
+          type: array
+          items:
+            type: string
+        max_depth:
+          type: integer
+        user_allocations: {}
+        replication_factor_min:
+          type: integer
+        name:
+          type: string
+        replication_factor_max:
+          type: integer
+        shard_size:
+          type: integer
+        type:
+          type: integer
+        cid:
+          $ref: '#/components/schemas/Cid'
+    
+    PinInfo:
+      type: object
+      properties:
+        peer_map:
+          type: object
+          additionalProperties:
+            $ref: '#/components/schemas/PeerMap'
+        cid:
+          $ref: '#/components/schemas/Cid'
+  
+    PeerMap:
+      type: object
+      properties:
+        peername:
+          type: string
+        peer:
+          type: string
+        error:
+          type: string
+        cid:
+          $ref: '#/components/schemas/Cid'
+        status:
+          type: string
+        timestamp:
+          type: string
+           
+    Error:
+      type: object
+      properties:
+        code:
+          type: integer
+        message:
+          type: string
+
+    Metric:
+      type: object
+      properties:
+        name:
+          type: string
+        peer:
+          type: string
+        value:
+          type: string
+        expire:
+          type: integer
+        valid:
+          type: boolean
+          
+  parameters:
```

Comment:
descriptions missing from most parameters

Reply:
I would add a description for progress, but it doesn't seem like it is being used. 
Not sure what I should write for shard size

Reply:
progress is used (but handled in the `addutils/` helper probably). `progress=false` disables streaming and returns an array of status at the end (progress=false actually makes the /add endpoint restful).

Reply:
you can leave sharding options undocumented, those are not used.

Reply:
sharding options are not used by cli, but they are still used by the apis.

> progress is used (but handled in the addutils/ helper probably). progress=false disables streaming and returns an array of status at the end (progress=false actually makes the /add endpoint restful).

I see `params.StreamChannels` doing this.https://github.com/ipfs/ipfs-cluster/blob/e523215ee29b833c7cfdfb9315dd4a825a292f12/adder/adderutils/adderutils.go#L59-L90

Reply:
sorry, you are right, we don't yet use progress for anything.